jobs/haproxy/templates/haproxy.ctmpl.erb

```diff
@@ -55,24 +55,21 @@ frontend http-in
 <% end %>
 <% end %>
 
-<% if_p("ha_proxy.ssl_pem") do %>
-frontend https-in
-    mode http
+<% if p("ha_proxy.ssl_pem") %>
+# To implement ELB+SSL+ProxyProtocol, we had to split this frontend in two
+# parts because HAProxy expects the ProxyProtocol header outside of the
+# SSL stream, meanwhile ELB sends it inside. The two configurations are:
+#
+#  * A plain TCP+SSL listener to do the SSL termination, and sends to...
+#  * ...a HTTP frontend which will parse the proxy protocol header.
+#
+# More info here: https://serverfault.com/questions/775010/aws-elb-with-ssl-backend-adds-proxy-protocol-inside-ssl-stream
+listen https-in
+    mode tcp
+    option splice-auto
     bind :443 ssl crt /var/vcap/jobs/haproxy/config/cert.pem no-sslv3 ciphers <%= p("ha_proxy.ssl_ciphers") %>
     capture request header Host len 128
```

Comment:
Auto answer: yes we do but it's fine, https-in listener is in tcp mode so haproxy logs only:
```
Jan 11 16:48:35 localhost haproxy[4462]: Connect from 10.0.0.170:15812 to 10.0.48.101:443 (https-in/TCP)
```